### ISSUE TITLE
 response: without specific logs or data related to the ROS 2 mobile robot's perception

### ISSUE THREAD

**Issue by saxenam06** [Dsobh/explainable_ros#6]
Hello, Many thanks for your work and open sourcing the repo. I got this running however with a below response from ros2 service call /question explicability_msgs/srv/Question "{'question': 'What is happening?'}". May be I am still missing something. Many thanks for your effort.

requester: making request: explicability_msgs.srv.Question_Request(question='What is happening?')

response:
explicability_msgs.srv.Question_Response(answer=" Without specific logs or data related to the ROS 2 mobile robot's perception, it's impossible for me to determine if obstacles are detected or describe the image-to-text correlation. I need the necessary input to analyze the situation and provide an explanation.").

Results from ros2 launch explicability_bringup explicability_ros.launch.py is as below: 

[llama_node-2] [INFO] [1735838270.835121586] [llama.llama_node]: Prompt received:
[llama_node-2] You are an explainability AI tool for ROS 2 mobile robots. Your goal is to detect possible obstacles. You have to comparing obstacle detected logs with image-to-text logs. You have to interpret the robot's data.
[llama_node-2]
[llama_node-2] Relevant data:
[llama_node-2] None
[llama_node-2]
[llama_node-2] USER: Given the context information and not prior knowledge, answer the query: What is happening?
[llama_node-2] ASSISTANT:
[llama_node-2] [INFO] [llama.cpp:generate_response:665] Sampler params:         repeat_last_n = 64, repeat_penalty = 1.000, frequency_penalty = 0.000, presence_penalty = 0.000
[llama_node-2]  dry_multiplier = 0.000, dry_base = 1.750, dry_allowed_length = 2, dry_penalty_last_n = -1
[llama_node-2]  top_k = 40, top_p = 0.950, min_p = 0.050, xtc_probability = 0.000, xtc_threshold = 0.100, typical_p = 1.000, temp = 0.000
[llama_node-2]  mirostat = 0, mirostat_lr = 0.100, mirostat_ent = 5.000
[llama_node-2] [INFO] [llama.cpp:generate_response:666] Sampler constr: logits -> logit-bias -> penalties -> dry -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
[llama_node-2] [INFO] [llama.cpp:generate_response:669] Prompt tokens:
[llama_node-2]  You are an explainability AI tool for ROS 2 mobile robots. Your goal is to detect possible obstacles. You have to comparing obstacle detected logs with image-to-text logs. You have to interpret the robot's data.
[llama_node-2]
[llama_node-2] Relevant data:
[llama_node-2] None
[llama_node-2]
[llama_node-2] USER: Given the context information and not prior knowledge, answer the query: What is happening?
[llama_node-2] ASSISTANT:
[llama_node-2] [INFO] [llama.cpp:generate_response:672] Starting Response Generation
               [INFO] [llama.cpp:generate_response:697] Partial stopping word found
               [INFO] [llama.cpp:find_stop:805] Stopping with EOS
[llama_node-2] [INFO] [llama.cpp:generate_response:722] Finish Response Generation
[llama_node-2] llama_perf_sampler_print:    sampling time =       1.88 ms /   142 runs   (    0.01 ms per token, 75411.58 tokens per second)
[llama_node-2] llama_perf_context_print:        load time =     904.53 ms
[llama_node-2] llama_perf_context_print: prompt eval time =   23818.00 ms /    86 tokens (  276.95 ms per token,     3.61 tokens per second)
[llama_node-2] llama_perf_context_print:        eval time =   22536.56 ms /    55 runs   (  409.76 ms per token,     2.44 tokens per second)
[llama_node-2] llama_perf_context_print:       total time =  106970.46 ms /   141 tokens

**Comment by saxenam06** [comment]
explicability_node needs some messages to be publshed on '/rosout' topic to start embedding and give RAG based responses. this was not happening by default by just launching explicabiltiy_node and llama_node. I solved this by just publishing some messages (for e.g. Twist message) on the '/rosout' topic.
